src/Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG     PYVER=3
+FROM    python:${PYVER}
+
+# Enable unbuffered STDOUT logging
+ENV     PYTHONUNBUFFERED=1
+
+ARG     NODEVER=12.x
+RUN     curl -sL https://deb.nodesource.com/setup_${NODEVER} | bash - \
+        && echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" > /etc/apt/sources.list.d/google-cloud-sdk.list \
+        && curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+
+RUN     apt update && apt install -y \
+          google-cloud-sdk \
```

Comment:
Do we need all these Google Cloud references? As I understand it they are only needed for release so only core maintainers need them so should we stick them under a comment or the like so every developer is not lumped with them?

Reply:
> This looks great! Thanks very much for working on this. Very nicely explained (even for a complete Docker-novice like myself!).

Thanks! I did skip many details for the sake of brevity. For example, I did not go into the details of mounted volumes and mapped ports because those familiar with Docker would know these things.

> I've a few comments and also a question: is it possible to have a pre-built Docker image and save it in the repo? I'm thinking for the GitHub Actions there might be a good time saving for it having a pre-built image. Obviously it would need to be regenerated occasionally to dependencies up to date but still could be useful.

I thought about it, but realized that this is not a general purpose application that people would want to run independently as a service, so publishing a Docker image has little utility. That's why I did not optimize the image for running it as a service, instead, I kept it developer-friendly with all the necessary tools and packages included at the cost of longer build time and heavier image size on disk. Also, I did not add any metadata labels in it, that I would have, if I were to prepare it for distribution. That said, if a compelling utility emerges beyond developers pulling the image instead of building it locally, I am open to add a GH Workflow file to automatically build the image every time a commit or tag is pushed to the main branch and push the image to DockerHub.

Reply:
I added it so that those who need it and have the ability to use it can use it without manually installing the tool. Others, who do not need it can leave it alone in the image, it will not bother them. The point of this developer image is to make it as self-contained and friction-less as possible. I could have provided means to selectively install it using build args or multi-stage builds, but then the Dockerfile will have more build branches and not very easy to understand or maintain going forward. Additionally, we will have to document those variations in the build process, which will add more text about Docker than the main agenda of the repo.

That said, I have intentionally placed this instruction early in the Dockerfile to leverage layer caching, which means, successive builds of the image after code changes will not invalidate this layer and it will only be downloaded and installed once, unless something changes upstream.

Reply:
Hmmmm I still think we should rip it out. I don't think we'll be able to deploy via this Docker Image which means deployments will always need to run outside of Docker so it's a waste of time and space installing this inside of Docker.

Reply:
I have made some changes here, it will now only be installed when asked explicitly at image build time.

Reply:
Thanks! Resolving.

Reply:
To test this change, run the following:

```
$ docker image build -t webalmanac:nogc .
$ docker container run --rm -it webalmanac:nogc gcloud --version
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"gcloud\": executable file not found in $PATH": unknown.
$ docker image build -t webalmanac:gc --build-arg SKIPGC=false .
$ docker container run --rm -it webalmanac:gc gcloud --version
Google Cloud SDK 299.0.0
alpha 2020.06.26
beta 2020.06.26
bq 2.0.58
core 2020.06.26
gsutil 4.51
kubectl 2020.06.26
```

Reply:
I'll be perfectly honest with you and did not run this in Docker as don't feel I know enough about it and I intend to just develop locally for reasons given earlier. But did a manual review of the changes and am happy enough to merge if this makes life easier for others (yourself included).

Reply:
I knew you did not test it, that's why I provided the outcome of the variations controlled by a build arg. By the way, there is an undocumented feature that would allow us to build images with different version combinations of Python and Node for testing without any changes in the Dockerfile, simply by passing appropriate build args.